stake-pool/program/src/instruction.rs

```diff
@@ -421,6 +421,57 @@ pub enum StakePoolInstruction {
         #[allow(dead_code)]
         uri: String,
     },
+
+    /// (Staker only) Redelegate active stake on a validator, eventually moving it to another
+    ///
+    /// Internally, this instruction splits a validator stake account into its
+    /// corresponding transient stake account, redelegates it to an ephemeral stake
+    /// account, then merges that stake into the destination transient stake account.
+    ///
+    /// In order to rebalance the pool without taking custody, the staker needs
+    /// a way of reducing the stake on a stake account. This instruction splits
+    /// some amount of stake, up to the total activated stake, from the canonical
+    /// validator stake account, into its "transient" stake account.
+    ///
+    /// The instruction only succeeds if the source transient stake account and
+    /// ephemeral stake account do not exist.
+    ///
+    /// The amount of lamports to move must be at least twice rent-exemption
+    /// plus the minimum delegation amount. Rent-exemption is required for the
+    /// source transient stake account, and rent-exemption plus minimum delegation
+    /// is required for the destination ephemeral stake account.
+    ///
+    ///  0. `[]` Stake pool
+    ///  1. `[s]` Stake pool staker
+    ///  2. `[]` Stake pool withdraw authority
+    ///  3. `[w]` Validator list
+    ///  4. `[w]` Source canonical stake account to split from
+    ///  5. `[w]` Source transient stake account to receive split and be redelegated
+    ///  6. `[w]` Uninitialized ephemeral stake account to receive redelegation
+    ///  7. `[w]` Destination transient stake account to receive ephemeral stake by merge
+    ///  8. `[w]` Destination stake account to receive transient stake after activation
```

Comment:
```suggestion
    ///  8. `[]` Destination stake account to receive transient stake after activation
```
`AccountMeta::new_readonly(*destination_validator_stake, false),`